src/mscorlib/src/System/TimeSpan.cs

```diff
@@ -405,6 +405,42 @@ public String ToString(String format, IFormatProvider formatProvider)
             return t1.Add(t2);
         }
 
+        public static TimeSpan operator *(TimeSpan timeSpan, double factor)
+        {
+            if (double.IsNaN(factor))
+            {
+                throw new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"), nameof(factor));
+            }
+
+            double ticks = timeSpan.Ticks * factor;
+            if (ticks > long.MaxValue | ticks < long.MinValue)
+            {
+                throw new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
+            }
+
+            return FromTicks((long)ticks);
+        }
+
+        public static TimeSpan operator *(double factor, TimeSpan timeSpan) => timeSpan * factor;
+
+        public static TimeSpan operator /(TimeSpan timeSpan, double divisor)
+        {
+            double ticks = timeSpan.Ticks / divisor;
+            if (ticks > long.MaxValue || ticks < long.MinValue || double.IsNaN(ticks))
+            {
+                // Assuming exceptions are the rarer case, and since both IsNaN(divisor) and 0/0 can
+                // result in IsNaN(ticks), defer determining which exception until this branch has
+                // been taken.
+                throw double.IsNaN(divisor)
+                    ? new ArgumentException(Environment.GetResourceString("Arg_CannotBeNaN"), nameof(divisor))
+                    : (Exception)new OverflowException(Environment.GetResourceString("Overflow_TimeSpanTooLong"));
+            }
+
+            return FromTicks((long)ticks);
+        }
+
+        public static double operator /(TimeSpan t1, TimeSpan t2) => t1.Ticks / (double)t2.Ticks;
```

Comment:
Practically we're dividing longs as a matter of representation, but theoretically we're dividing a span of time by another. I think it's reasonable to be able to ask the question "how many zero-second periods in an hour?"